### ISSUE TITLE
Issues with Bitbucket Server/Stash usersnames containing '@'

### ISSUE THREAD

**Issue by idabrogie** [drone/drone#2162]
### with option closed registration
this is the output for admins when trying to add new users:
```
$ drone user add ida
Successfully added user ida
$ drone user add first.last@company.com
client error 400: Invalid User Login
```
### with option open registration 
usernames with '@' works fine. 
but users with '@' can't access drone through the cli.

#### drone version used: 0.8-rc3

Question posted in drone gitter channel earlier today.

**Comment by bradrydzewski** [comment]
When creating a user it must match the following regular expression
https://github.com/drone/drone/blob/master/model/user.go#L9

We would need to use a different regular expression for Bitbucket Server, since all other providers prevent the ampersand symbol and punctuation in usernames.

**Comment by idabrogie** [comment]
ok thank you for answering. but for us it works fine with stash usernames like 'firstname.lastname'. but according to the regular expression it should not work.

**Comment by bradrydzewski** [comment]
> ok thank you for answering. but for us it works fine with stash usernames like 'firstname.lastname'. but according to the regular expression it should not work.

the regular expression only applies when manually creating a user account (via the API endpoint). The system does not verify user accounts that come directly from the version control system, since it knows they are already valid.

I think the next step to resolve this issue is to submit a pull request that changes the regular expression at https://github.com/drone/drone/blob/master/model/user.go#L9

**Comment by bradrydzewski** [comment]
closed by acb6a8b02af3cd52b1283cd7a938a00f9f2e326b
